### PR TITLE
Fixing a bug where when you delete a cell with array cells it doesn't…

### PR DESCRIPTION
**Pull request by davidkircos**
… create the proper undo/redo transaction

### CODE REVIEW

File: src/grid/controller/sheetController.ts [hunk 1]
```diff
@@ -61,7 +61,7 @@ export class SheetController {
     // run statement and add reverse statement to transaction_in_progress
     const reverse_statement = StatementRunner(this.sheet, statement, this.app);
 
-    this.transaction_in_progress_reverse.statements.push(reverse_statement);
+    this.transaction_in_progress_reverse.statements.unshift(reverse_statement);
```
Comment (davidkircos) [inline src/grid/controller/sheetController.ts:64]: for a reverse transaction, statements need to be in the reverse order. 

Push puts statement at the end of the array. Unshift puts it at the beginning.

File: src/grid/actions/updateCellAndDCells.ts [hunk 2]
```diff
@@ -215,10 +215,11 @@ export const updateCellAndDCells = async (args: ArgsType) => {
     );
 
     // delete old array cells
-    array_cells_to_delete.forEach((cell) => {
+    array_cells_to_delete.forEach((aCell) => {
+      if (aCell.x === cell?.x && aCell.y === cell?.y) return; // don't delete the cell we just updated (it's in array_cells_to_output)
```
Comment (davidkircos) [inline src/grid/actions/updateCellAndDCells.ts:219]: cell was deleting itself

File: src/grid/controller/tests/basic.test.ts [hunk 3]
```diff
@@ -115,3 +115,70 @@ test('SheetController - code is saved from undo to redo', () => {
...
+  expect(sc.sheet.grid.getCell(0, 0)).toBeUndefined();
+
+  sc.redo();
+
+  expect(sc.sheet.grid.getCell(0, 0)?.value).toBe('hello');
+
+  sc.redo();
+
+  expect(sc.sheet.grid.getCell(0, 0)?.value).toBe('hello2');
+});
```
Comment (jimniels) [inline src/grid/controller/tests/basic.test.ts:184]: nice. great set of tests here

**Review by jimniels — APPROVED** [review]
No more undo/redo until you get what you were expecting ha

This is a nice change in the functionality of the app, but the tests add a lot too!

👍

**Pull request merged**